src/UnitTests/Projection/MapFromTest.cs

```diff
@@ -21,6 +21,18 @@ public void Should_not_fail()
         }
 
         [Fact]
+        public void Should_not_fail_Untyped()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<UserModel, UserDto>()
+                    .ForMember(dto => dto.FullName, opt => opt.MapFrom(src => src.LastName + " " + src.FirstName));
+            });
+
+            typeof(ArgumentNullException).ShouldNotBeThrownBy(() => config.ExpressionBuilder.GetMapExpression(typeof(UserModel), typeof(UserDto), null, null)); //no ArgumentNullException here
```

Comment:
Changing the optional param is a breaking change as far as I know, but if I'm touching the API there I can add an overload without the optional params (apparently it's usually called without any params at all)